source/AI.cpp

```diff
@@ -1605,6 +1717,45 @@ bool AI::FollowOrders(Ship &ship, Command &command) const
 
 
 
+void AI::MoveInFormation(Ship &ship, Command &command)
+{
+	shared_ptr<Ship> parent = ship.GetParent();
+	if(!parent)
+		return;
+
+	const Body *formationLead = parent.get();
+	const FormationPattern *pattern = ship.GetFormationPattern();
+
+	// First we retrieve the patterns that are formed around the parent.
+	auto &patterns = formations[formationLead];
+
+	// Find the existing FormationPositioner for the pattern, or add one if none exists yet.
+	auto insert = patterns.emplace(piecewise_construct,
+		forward_as_tuple(pattern),
+		forward_as_tuple(formationLead, pattern));
+
+	// Set an iterator to point to the just found or emplaced value.
+	auto it = insert.first;
+
+	// Aggressively try to match the position and velocity for the formation position.
+	const double positionDeadband = ship.Radius() * 1.25;
+	constexpr double VELOCITY_DEADBAND = .1;
+	bool inPosition = MoveTo(ship, command, it->second.Position(&ship), formationLead->Velocity(), positionDeadband,
+		VELOCITY_DEADBAND);
+
+	// If we match the position and velocity, then also match the facing angle.
```

Comment:
This should have some tolerance built in as well. I tried getting a dozen Deep Rivers into a line formation, and they were... vibrating? Like they were trying to match the exact angle of my flagship, but failing, and on their scale, it was rather jarring.

Reply:
Agreed, some tolerance here would be nice.

Reply:
I added a tolerance of 3 degrees (in both directions). I expect that this fixes the jarring, but let me know if it is not enough.